File: CCO C++/cco07p2_snowflakes/CCO07P2.cpp
/*
 * CCO07P2.cpp
 *
 *  Created on: Apr 25, 2017
 *      Author: Wesley Leung
 */

#include <bits/stdc++.h>
#include <algorithm>

using namespace std;

unordered_set<string> hs;
int lengths[6];
int N;

int main() {
    scanf("%d", &N);
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < 6; j++) {
            scanf("%d", &lengths[j]);
        }
        sort(begin(lengths), end(lengths));
        string str = "";
        for (int j = 0; j < 6; j++) {
            str += lengths[j];
        }
        hs.insert(str);
    }
    if (hs.size() == N) printf("No two snowflakes are alike.");
    else printf("Twin snowflakes found.");
    return 0;
}

File: CCO Java/cco07p2_snowflakes/CCO07P2.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.util.Arrays;
import java.util.HashSet;
import java.util.StringTokenizer;

public class CCO07P2 {
    private static CCO07P2 o = new CCO07P2();
    public class Reader {
        private BufferedReader in;
        private StringTokenizer st;
        
        public Reader(InputStream inputStream) {
            in = new BufferedReader(new InputStreamReader(inputStream));
        } // Reader InputStream constructor
        
        public Reader(String fileName) throws FileNotFoundException {
            in = new BufferedReader(new FileReader(fileName));
        } // Reader String constructor

        public String next() throws IOException {
            while (st == null || !st.hasMoreTokens()) {
                st = new StringTokenizer(in.readLine().trim());
            } // while
            return st.nextToken();
        } // next method
        
        public long nextLong() throws IOException {
            return Long.parseLong(next());
        } // nextLong method
        
        public int nextInt() throws IOException {
            return Integer.parseInt(next());
        } // nextInt method
        
        public double nextDouble() throws IOException {
            return Double.parseDouble(next());
        } // nextDouble method
        
        public String nextLine() throws IOException {
            return in.readLine().trim();
        } // nextLine method
    } // Reader class
    
    public static void main(String[] args) throws IOException {
        Reader in = o.new Reader(System.in);
        PrintWriter out = new PrintWriter(new OutputStreamWriter(System.out));
        HashSet<String> hs = new HashSet<String>();
        int N = in.nextInt();
        for (int i = 0; i < N; i++) {
            int[] lengths = new int[6];
            for (int j = 0; j < lengths.length; j++) { 
                lengths[j] = in.nextInt();
            }
            Arrays.sort(lengths);
            String str = "";
            for (int j = 0; j < lengths.length; j++) { 
                str += lengths[j];
            }
            hs.add(str);
        }
        if (hs.size() == N) out.println("No two snowflakes are alike.");
        else out.println("Twin snowflakes found.");
        out.close();
    }
}
